libs/testing/src/jobs.service.ts

```diff
@@ -1,262 +1,238 @@
+   * | workflow queue > subscriber queue > standard queue |
+   * |----------------|------------------|----------------|
+   *
+   * @remarks
+   * This is useful in testing when you want the trigger to be asserted in specific parts of the execution.
+   * For example, you can wait for the subscriber queue to be completed and then assert that the trigger was sent to the standard queue.
+   */
+  public async waitForSubscriberQueueCompletion(maxWaitTime = 10000) {
+    return this.waitQueueUntil(
+      ({ activeSubscriberJobsCount, waitingSubscriberJobsCount }) =>
+        activeSubscriberJobsCount + waitingSubscriberJobsCount > 0,
+      maxWaitTime
+    );
+  }
 
-    // todo check of waitForStorageImmediateJobCompletion is needed as it should be equivalent to waitForQueueImmediateJobCompletion
-    await this.waitForQueueImmediateJobCompletion();
-    await this.waitForStorageImmediateJobCompletion();
+  /**
+   * Wait for all jobs to be completed from the standard Redis queue
+   *
+   * |----------------|------------------|----------------|
+   * | workflow queue > subscriber queue > standard queue |
+   * |----------------|------------------|----------------|
+   *
+   * @remarks
+   * This is useful in testing when you want the trigger to be asserted in specific parts of the execution.
+   * For example, you can wait for the standard queue to be completed and then assert agains the stage of the job in Mongo
+   */
+  public async waitForStandardQueueCompletion(maxWaitTime = 10000) {
+    return this.waitQueueUntil(
+      ({ activeStandardJobsCount, waitingStandardJobsCount }) => activeStandardJobsCount + waitingStandardJobsCount > 0,
+      maxWaitTime
+    );
+  }
 
-    while (hasMoreDelayedJobs && iterationCount < MAX_ITERATIONS) {
-      hasMoreDelayedJobs = await this.promoteDelayedJobs();
+  public async waitQueueUntil(
+    cb: (metrics: Awaited<ReturnType<typeof this.getQueueMetrics>>) => boolean,
+    maxWaitTime = 10000
+  ) {
+    const startTime = Date.now();
+    const safeMaxWaitTime = Math.min(200, maxWaitTime);
 
-      if (!hasMoreDelayedJobs) {
-        continue;
-      }
+    let queueMetrics: Awaited<ReturnType<typeof this.getQueueMetrics>>;
 
-      // todo check of waitForStorageImmediateJobCompletion is needed as it should be equivalent to waitForQueueImmediateJobCompletion
-      await this.waitForQueueImmediateJobCompletion();
-      await this.waitForStorageImmediateJobCompletion();
-      iterationCount += 1;
-    }
+    do {
+      await setTimeout(100);
+      queueMetrics = await this.getQueueMetrics();
+    } while (cb(queueMetrics) && Date.now() - startTime < safeMaxWaitTime);
+  }
 
-    if (iterationCount >= MAX_ITERATIONS) {
-      // eslint-disable-next-line no-console
-      console.warn(
-        'Max iterations reached while processing delayed jobs. This might indicate an infinite loop in job creation.'
-      );
-    }
+  public async runStandardQueueDelayedJobsImmediately() {
```

Comment:
interesting that you managed to make all the tests pass with only promote of standard, while debugging the e2e on my pr i had cases where the subscriber queue was my bottle neck that i needed wait for.